gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -275,6 +278,22 @@ public GsonBuilder enableComplexMapKeySerialization() {
     return this;
   }
 
+  /**
+   * Configures Gson to deserialize duplicate map keys. Only the value of last entry with the same key will be used, previous values
+    * will be discarded. By default, Gson throws a {@link JsonSyntaxException} when a key occurs more than once.
+   *
+   * <p>Note that enabling support for duplicate maps keys is discouraged because it can make an application less secure.
+   * When an application interacts with other components using different JSON libraries, they might treat duplicate keys
+   * differently, allowing an attacker to circumvent security checks.
+   *
+   * @return a reference to this {@code GsonBuilder} object to fulfill the "Builder" pattern
+   * @since 2.8
+   */
+  public GsonBuilder enableDuplicateMapKeysDeserialization() {
```

Comment:
Sorry, I missed that `GsonBuilder` already has `enableComplexMapKeySerialization` ("...Key<del>s</del>"). So maybe for consistency it would be good to use "...Key<del>s</del>" here as well.
```suggestion
  public GsonBuilder enableDuplicateMapKeyDeserialization() {
```